bamt/builders/evo_builder.py

```diff
@@ -0,0 +1,195 @@
+from datetime import timedelta
+
+from pandas import DataFrame
+
+from bamt.builders.builders_base import BaseDefiner
+from bamt.utils import EvoUtils as evo
+
+from golem.core.adapter import DirectAdapter
+from golem.core.dag.verification_rules import has_no_cycle, has_no_self_cycled_nodes
+from golem.core.optimisers.genetic.gp_optimizer import EvoGraphOptimizer
+from golem.core.optimisers.genetic.gp_params import GPAlgorithmParameters
+from golem.core.optimisers.genetic.operators.crossover import CrossoverTypesEnum
+from golem.core.optimisers.genetic.operators.inheritance import GeneticSchemeTypesEnum
+from golem.core.optimisers.objective import Objective, ObjectiveEvaluate
+from golem.core.optimisers.optimization_parameters import GraphRequirements
+from golem.core.optimisers.optimizer import GraphGenerationParams
+from golem.core.optimisers.genetic.operators.selection import SelectionTypesEnum
+
+from typing import Dict, Optional
+
+
+class EvoDefiner(BaseDefiner):
+    """
+    Object that might take additional methods to decompose structure builder class
+    """
+    def __init__(self, data: DataFrame, descriptor: Dict[str, Dict[str, str]],
+                 regressor: Optional[object] = None):
+
+        super().__init__(data, descriptor, regressor)
+
+
+class EvoStructureBuilder(EvoDefiner):
+    """
+    This class uses an evolutionary algorithm based on GOLEM to generate a Directed Acyclic Graph (DAG) that represents
+    the structure of a Bayesian Network.
+
+    Attributes:
+        data (DataFrame): Input data used to build the structure.
+        descriptor (dict): Descriptor describing node types and signs.
+        regressor (object): A regression model for continuous nodes.
+        has_logit (bool): Indicates whether a logit link function should be used.
+        use_mixture (bool): Indicates whether a mixture model should be used.
+    """
+    def __init__(
+            self,
+            data: DataFrame,
+            descriptor: Dict[str, Dict[str, str]],
+            regressor: Optional[object],
+            has_logit: bool,
+            use_mixture: bool):
+        super(
+            EvoStructureBuilder,
+            self).__init__(
+            data=data,
+            descriptor=descriptor,
+            regressor=regressor)
```

Comment:
Why in "build" we can specify both "classifier" and "regressor" and in "EvoStructureBuilder" only "regressor"?

Reply:
Classifier is already there, it is being taken from ``build`` function